src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java

```diff
@@ -84,6 +87,14 @@ public static StructuredSyslogMessage fromString(
 
     private void  deserialize(final String stringMessage) {
 
+        // Check if the RFC5424 MSGID and STRUCTURED-DATA fields are empty.
+        // This avoids throwing an exception and also strips the "- - " from the message.
+        // See: https://github.com/Graylog2/graylog2-server/issues/1161
+        if (stringMessage.startsWith(EMPTY_STRUCTURED_DATA_PREFIX)) {

```

Comment:
You're correct. I was wrongly assuming that RFC 3164 messages could also be an input for that method.

I've added a test in commit 6d1294cf7ec862aa62ba3a186b10e2663d447311 to test my imagined corner-case specifically.